src/grappler/ngraph_optimizer.cc

```diff
@@ -166,8 +191,30 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
     DumpGraphs(graph, idx, "unmarked", "Unmarked Graph");
   }
 
+  // Get backend + its configurations, to be attached to the nodes
```

Comment:
Can you please add a comment about the order or precedence:

BM < Env var <  rewriter config

Is the precedence intentional, or do you want to refactor the if-else to make the precedence BM < rewriter config < Env var